### PR TITLE
Add Flowery TTS as source

### PR DESCRIPTION
**Pull request by bachtran02**
Add support for Flowery TTS - [Flowery TTS Support #66 ](https://github.com/topi314/LavaSrc/issues/66)

### CODE REVIEW

File: plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/LavaSrcPlugin.java [hunk 1]
```diff
@@ -65,6 +68,20 @@ public AudioPlayerManager configure(AudioPlayerManager manager) {
 			log.info("Registering Yandex Music audio source manager...");
 			manager.registerSourceManager(new YandexMusicSourceManager(this.yandexMusicConfig.getAccessToken()));
 		}
+		if (this.sourcesConfig.isFloweryTTS()) {
```
Comment (topi314) [inline plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/LavaSrcPlugin.java:71]: can you please validate this in the constructor and not here?

File: main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    @Override
+    public AudioItem loadItem(AudioPlayerManager manager, AudioReference reference){
+
+        if (reference.identifier.startsWith(TTS_PREFIX)) {
+            final String query = reference.identifier.substring(TTS_PREFIX.length());
+            final URI uri = this.buildURI(query);
+
+            return new FloweryTTSAudioTrack(
+                    new AudioTrackInfo(
+                            "text-to-speech",
```
Comment (topi314) [inline main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java:81]: the track title should make sense, and not be a static string

File: main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                            "FloweryTTS",
+                            Units.CONTENT_LENGTH_UNKNOWN,
+                            "no-id",
```
Comment (topi314) [inline main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java:84]: the identifier should be unique per track

**Review by topi314 — CHANGES_REQUESTED** [review]

File: main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java (hunk 3 above)
Comment (bachtran02) [reply main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java]: fixed in latest commit

File: main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java (hunk 2 above)
Comment (bachtran02) [reply main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java]: fixed in latest commit

File: plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/LavaSrcPlugin.java (hunk 1 above)
Comment (bachtran02) [reply plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/LavaSrcPlugin.java:71]: fixed in latest commit

File: application.yml.example [hunk 4]
```diff
@@ -26,6 +26,11 @@ plugins:
       masterDecryptionKey: "your master decryption key" # the master key used for decrypting the deezer tracks. (yes this is not here you need to get it from somewhere else)
     yandexmusic:
       accessToken: "your access token" # the token used for accessing the yandex music api. See https://github.com/TopiSenpai/LavaSrc#yandex-music
+    flowerytts:
```
Comment (topi314) [inline application.yml.example:29]:
```suggestion
    flowery_tts:
```

File: main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                    new AudioTrackInfo(
+                        queryUri.getAuthority(),
+                        "FloweryTTS",
+                        Units.CONTENT_LENGTH_UNKNOWN,
+                        requestUri.toString(),
+                        false,
+                        null),
+                    this);
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
```
Comment (topi314) [inline main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java:116]:
```suggestion
                throw new RuntimeException(e);

```

File: plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/FloweryTTSConfig.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.stereotype.Component;
+
+@ConfigurationProperties(prefix = "plugins.lavasrc.flowerytts")
+@Component
+public class FloweryTTSConfig {
+
+    private final int SILENCE_MIN = 0;
+    private final int SILENCE_MAX = 10000;
+    private final float SPEED_MIN = 0.5f;
+    private final float SPEED_MAX = 10;
```
Comment (topi314) [inline plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/FloweryTTSConfig.java:13]:
```suggestion
    private static final int SILENCE_MIN = 0;
    private static final int SILENCE_MAX = 10000;
    private static final float SPEED_MIN = 0.5f;
    private static final float SPEED_MAX = 10;

```

File: plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/FloweryTTSConfig.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    public void setTranslate(boolean translate) {
+        this.translate = translate;
+    }
+
+    public int getSilence(){
+        return this.silence;
+    }
+
+    public void setSilence(int silence){
+        this.silence = silence > SILENCE_MAX ? SILENCE_MAX : silence < SILENCE_MIN ? SILENCE_MIN : silence;
```
Comment (topi314) [inline plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/FloweryTTSConfig.java:41]: please use `Math.min` & `Math.max`

File: plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/FloweryTTSConfig.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public float getSpeed(){
+        return this.speed;
+    }
+
+    public void setSpeed(float speed){
+        this.speed = speed > SPEED_MAX ? SPEED_MAX : speed < SPEED_MIN ? SPEED_MIN : speed;
```
Comment (topi314) [inline plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/FloweryTTSConfig.java:49]: please use `Math.min` & `Math.max`

File: plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/LavaSrcPlugin.java [hunk 9]
```diff
@@ -65,6 +68,20 @@ public AudioPlayerManager configure(AudioPlayerManager manager) {
...
+			log.info("Registering Flowery TTS audio source manager...");
+			var floweryTTSSourceManager = new FloweryTTSSourceManager(this.floweryTTSConfig.getVoice());
+			if (this.floweryTTSConfig.getTranslate() == true){
```
Comment (topi314) [inline plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/LavaSrcPlugin.java:74]:
```suggestion
			if (this.floweryTTSConfig.getTranslate()){
```

File: plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/FloweryTTSConfig.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
+package com.github.topisenpai.lavasrc.plugin;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@ConfigurationProperties(prefix = "plugins.lavasrc.flowerytts")
```
Comment (topi314) [inline plugin/src/main/java/com/github/topisenpai/lavasrc/plugin/FloweryTTSConfig.java:6]:
```suggestion
@ConfigurationProperties(prefix = "plugins.lavasrc.flowery_tts")

```

File: main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+                        .filter((p) -> pair.getName().equals(p.getName()))
+                        .map(NameValuePair::getValue)
+                        .findFirst()
+                        .orElse(pair.getValue())
+                    );
+                }
+
+                return new FloweryTTSAudioTrack(
+                    new AudioTrackInfo(
+                        queryUri.getAuthority(),
```
Comment (topi314) [inline main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java:108]: this should be `tts://bla?key=value`, just assemble the api url once you acutally need it

File: main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+    private static final Logger log = LoggerFactory.getLogger(FloweryTTSSourceManager.class);
+
+    private final String voice;
+    private boolean translate = false;
+    private int silence = 0;
+    private float speed = 1;
+
+    private final HttpInterfaceManager httpInterfaceManager = HttpClientTools.createDefaultThreadLocalManager();
+
+    public FloweryTTSSourceManager(String voice) {
```
Comment (topi314) [inline main/src/main/java/com/github/topisenpai/lavasrc/flowerytts/FloweryTTSSourceManager.java:50]: lets use a default voice too & remove the voice param

**Review by topi314 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**